api/CSVHandler.py

```diff
@@ -50,3 +48,44 @@ def test_split_train_test():
     train, test = split_train_test(data)
     assert np.shape(train) == (40, 4)
     assert np.shape(test) == (27, 4)
+
+def create_graph_nodes(data):
+    for line in data:
+        if (line[0].isdigit()) and ("->" not in line):
+            node = line.split("\"")[1].split("\\n")
+            node_id = int(line[0])
+            gini = ""
+            samples = ""
+            values = ""
+            expression = ""
+            for attr in node:
+                if 'gini' in attr:
+                    gini = float(attr.split(" ")[-1])
+                elif 'samples' in attr:
+                    samples = float(attr.split(" ")[-1])
+                elif 'value' in attr:
+                    values = attr.split("[")[-1].replace("]", "").replace(" ", "").split(",")
+                    values = list(map(int, values))
```

Comment:
Very cool! Python prefers comprehensions, which are really neat and clear but not quite as powerful as maps. Here's how this would work in a list comprehension (there are also dictionary comprehensions and generators are a close cousin to comprehensions).

`values = [int(v) for v in values]`